fbw-a380x/src/wasm/systems/a380_systems/src/air_conditioning.rs

```diff
@@ -40,131 +55,2155 @@ impl A380AirConditioning {
     ) {
-        self.a380_air_conditioning_system.update(
+        self.pressurization_updater.update(context);
+
+        self.a320_air_conditioning_system.update(
             context,
             adirs,
-            &self.a380_cabin,
+            &self.a320_cabin,
             engines,
             engine_fire_push_buttons,
             pneumatic,
             pneumatic_overhead,
-            pressurization,
+            &self.a320_pressurization_system,
             pressurization_overhead,
             lgciu,
         );
-        self.a380_cabin.update(
-            context,
-            &self.a380_air_conditioning_system,
-            &self.a380_air_conditioning_system,
-            pressurization,
-        );
+
+        for cur_time_step in self.pressurization_updater {
+            self.a320_cabin.update(
+                &context.with_delta(cur_time_step),
+                &self.a320_air_conditioning_system,
+                lgciu,
+                &self.a320_pressurization_system,
+            );
+
+            // Temporary array until pressurization is done for A380
+            let temp_engines = [engines[0], engines[1]];
+
+            self.a320_pressurization_system.update(
+                &context.with_delta(cur_time_step),
+                adirs,
+                pressurization_overhead,
+                temp_engines,
+                lgciu,
+                &self.a320_cabin,
+            );
+        }
+    }
+
+    pub fn mix_packs_air_update(&mut self, pack_container: &mut [impl PneumaticContainer; 2]) {
+        self.a320_air_conditioning_system
+            .mix_packs_air_update(pack_container);
     }
 }
 
 impl PackFlowControllers<3, 4> for A380AirConditioning {
     fn pack_flow_controller(&self, pack_id: Pack) -> PackFlowController<3, 4> {
-        self.a380_air_conditioning_system
+        self.a320_air_conditioning_system
             .pack_flow_controller(pack_id)
     }
 }
 
 impl SimulationElement for A380AirConditioning {
     fn accept<T: SimulationElementVisitor>(&mut self, visitor: &mut T) {
-        self.a380_cabin.accept(visitor);
-        self.a380_air_conditioning_system.accept(visitor);
+        self.a320_cabin.accept(visitor);
+        self.a320_air_conditioning_system.accept(visitor);
+        self.a320_pressurization_system.accept(visitor);
 
         visitor.visit(self);
     }
 }
 
-struct A380Cabin {
-    cabin_zone: [CabinZone<2>; 3],
+struct A320Cabin {
+    passenger_rows_id: [Option<Vec<VariableIdentifier>>; 3],
+    fwd_door_id: VariableIdentifier,
+    rear_door_id: VariableIdentifier,
+
+    fwd_door_is_open: bool,
+    rear_door_is_open: bool,
+    number_of_passengers: [u8; 3],
+    cabin_air_simulation: CabinAirSimulation<A320PressurizationConstants, 3>,
 }
 
-impl A380Cabin {
-    // TODO: Improve volume according to specs
-    const A380_CABIN_VOLUME_CUBIC_METER: f64 = 200.; // m3
-    const A380_COCKPIT_VOLUME_CUBIC_METER: f64 = 10.; // m3
+impl A320Cabin {
```

Comment:
```suggestion
impl A320Cabin {
```
Shouldn't naming still be A380

Reply:
Yeah I left it as A320 to signify that it still hasn't been adapted to the A380. It's just a naming thing though so I can change it if you think it's better.